src/main/dataCenter/index.js

```diff
@@ -0,0 +1,162 @@
+import fs from 'fs'
+import path from 'path'
+import EventEmitter from 'events'
+import { BrowserWindow, ipcMain, dialog } from 'electron'
+import schema from './schema'
+import Store from 'electron-store'
+import log from 'electron-log'
+import { ensureDirSync } from '../filesystem'
+import { IMAGE_EXTENSIONS } from '../config'
+
+const DATA_CENTER_NAME = 'dataCenter'
+
+class DataCenter extends EventEmitter {
+  constructor (paths) {
+    super()
+
+    const { dataCenterPath, userDataPath } = paths
+    this.dataCenterPath = dataCenterPath
+    this.userDataPath = userDataPath
+    this.hasDataCenterFile = fs.existsSync(path.join(this.dataCenterPath, `./${DATA_CENTER_NAME}.json`))
+    this.store = new Store({
+      schema,
+      name: DATA_CENTER_NAME
+    })
+
+    this.init()
+  }
+  init () {
+    const defaltData = {
+      imageFolderPath: path.join(this.userDataPath, 'images/'),
+      screenshotFolderPath: path.join(this.userDataPath, 'screenshot/'),
+      webImages: [],
+      cloudImages: [],
+      currentUploader: 'smms',
+      imageBed: {
+        github: {
+          token: '',
```

Comment:
We should store sensitive information using `node-keytar` and never on disk!

Reply:
I'll add this in next commits.

Reply:
> In `node-keytar` readme file.
>Currently this library uses libsecret so you may need to install it before running npm install.

Can you help handle this, and help test it on Windows and Linux?

Reply:
`libsecret` should be installed on every Linux distro. I have to update the flatpak package later because Mark Text runs inside a sandbox. AppImage should be fine too. You only need to update `CONTRIBUTING.md # Build Instructions --> Additional development dependencies on Linux`.

Reply:
And Travis need an update:

```diff
diff --git a/.travis.yml b/.travis.yml
index f3a62e03..37076b99 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -31,6 +31,8 @@ addons:
     # atom/keyboard-layout
     - libx11-dev
     - libxkbfile-dev
+    # atom/node-keytar
+    - libsecret-1-dev

 before_install:
   - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sudo apt-get -qq update ; fi
```